### PR TITLE
Fix faulty default `WHERE` clause & column checks for joins on `MorphTo` relation

### LINKED ISSUE #160: Faulty default `WHERE` clause & `joinRelationship()` on morphable `MorphTo` has errors
Consider the set-up as in issue #147 

If you run:
```php
OrderItem::joinRelationship('orderable', morphable: Product::class)->getQuery()->toSQL()
```

It throws a deprecation error (using PHP 8.1.10):
```
DEPRECATED  str_contains(): Passing null to parameter #1 ($haystack) of type string is deprecated in vendor\laravel\framework\src\Illuminate\Databas
e\Eloquent\Model.php on line 569.
```

This is because the column variable, that is being checked there, is `null` instead of a proper name.

I suspect this empty column is from a (faulty) `WHERE` clause that is present by default on the relation's base query:
```php
OrderItem::getModel()->orderable()->getBaseQuery()->wheres
/*
  Returns:
  [
    [
      "type" => "Null",
      "column" => "order_items.",
      "boolean" => "and",
    ],
  ]
*/
```

### PR DESCRIPTION
**Pull request by vHeemstra**
Fixes #160 

For info, see the comments with the commits

### CODE REVIEW

File: src/Mixins/RelationshipsExtraMethods.php
Comment (vHeemstra) [inline src/Mixins/RelationshipsExtraMethods.php:1]: Unlike `MorphOneOrMany`, the `MorphTo` relation does not have these methods:
* `getQualifiedMorphType()`
* `getExistenceCompareKey()`

So we use the next best thing for these.

**Review by vHeemstra — COMMENTED** [review]
Unlike `MorphOneOrMany`, the `MorphTo` relation does not have these methods:
* `getQualifiedMorphType()`
* `getExistenceCompareKey()`

So we use the next best thing for these.

**Comment by luisdalmolin** [comment]
As you can see, this is not the right fix for the issue as it's breaking a couple of tests. If you can push the repo or some gists with the code I can try to replicate.

**Comment by vHeemstra** [comment]
> As you can see, this is not the right fix for the issue as it's breaking a couple of tests. If you can push the repo or some gists with the code I can try to replicate.

See  https://github.com/kirschbaum-development/eloquent-power-joins/issues/160#issuecomment-1783971619

**Comment by luisdalmolin** [comment]
I pushed #164 to fix this, closing this one. But thanks for the report!

**Pull request closed without merging**